asf_smart_meter_exploration/pipeline/data_aggregation.py

```diff
@@ -0,0 +1,177 @@
+# File: asf_smart_meter_exploration/pipeline/data_aggregation.py
+"""
+Functions to process household smart meter data into various formats for clustering.
+"""
+
+import numpy as np
+import holidays
+
+from asf_smart_meter_exploration.getters.get_processed_data import (
+    get_meter_data,
+    get_household_data,
+)
+
+
+meter_data = get_meter_data()
+
+
+def get_average_usage(meter_data=meter_data, normalised=False, cumulative=False):
+    """For each household, get average usage for each half-hour of the day.
+
+    Args:
+        meter_data (pd.DataFrame, optional): _description_. Defaults to meter_data.
+        normalised (bool, optional): Whether to normalise the readings and return
+            the proportion used in each half-hour rather than the amount in kWh.
+            Defaults to False.
+        cumulative (bool, optional): Whether to return the cumulative sums of the values instead.
+            Defaults to False.
+
+    Returns:
+        pd.DataFrame: Average usage data.
+    """
+
+    hh_averages = meter_data.groupby("time").mean(numeric_only=True).dropna(axis=1).T
```

Comment:
I think the missing values are genuine errors rather than no electricity being used. But agree that we should look out for this in the future and more rigorously check proportions of missing values (I think this is relatively clean/complete data as far as smart meter data goes so we will probably have to do a lot more work on this when we get a different dataset)